eslint.config.mjs

```diff
@@ -0,0 +1,391 @@
+      ],
+    },
+  },
+  {
+    files: ['packages/orchestration/src/exos/**'],
+
+    rules: {
+      'no-restricted-syntax': [
+        'error',
+        {
+          selector:
+            'FunctionExpression[async=true]:not(Property[key.name="connectionHandler"] > ObjectExpression > Property[key.name=/^(onOpen|onClose)$/] > FunctionExpression[async=true])',
+          message: 'Non-immediate functions must return vows, not promises',
+        },
+        {
+          selector: 'ArrowFunctionExpression[async=true]',
+          message: 'Non-immediate functions must return vows, not promises',
+        },
+        {
+          selector: "Identifier[name='callWhen']",
+          message:
+            'callWhen wraps the function in a promise; instead immediately return a vow',
+        },
+        {
+          selector: "Identifier[name='heapVowE']",
+          message:
+            'heapVowE shortens vows to promises; instead use `E` from `@endo/far` with `watch` from durable vowTools',
+        },
+        {
+          selector: "Identifier[name='heapVowTools']",
+          message:
+            'heapVowTools are not durable; instead use `prepareVowTools` with a durable zone',
+        },
+      ],
+    },
+  },
+  {
+    // Allow "loan" contracts to mention the word "loan".
+    files: ['packages/zoe/src/contracts/loan/*.js'],
+
+    rules: {
+      'no-restricted-syntax': ['error', ...deprecatedTerminology.loanContract],
+    },
+  },
+  {
+    files: ['**/*.ts'],
+
+    rules: {
+      'jsdoc/require-param-type': 'off',
+      'no-undef': 'off',
+    },
+  },
+  {
+    files: ['**/*.d.ts'],
+
+    rules: {
+      'no-redeclare': 'off',
+    },
+  },
+  ...compat
+    .extends('plugin:@typescript-eslint/disable-type-checked')
+    .map(config => ({
+      ...config,
+      files: [
+        '**/exported.*',
+        '**/types-index.*',
+        '**/types-ambient.*',
+        '**/types.*',
+      ],
+    })),
+  {
+    files: ['**/*.html'],
+
+    languageOptions: {
+      ecmaVersion: 5,
+      sourceType: 'script',
+
+      parserOptions: {
+        project: false,
+      },
+    },
+  },
+  ...compat
+    .extends('plugin:@typescript-eslint/disable-type-checked')
+    .map(config => ({
+      ...config,
+      files: ['a3p-integration/**'],
+    })),
+  {
+    files: ['a3p-integration/**'],
+
+    languageOptions: {
+      ecmaVersion: 5,
+      sourceType: 'script',
+
+      parserOptions: {
+        useProjectService: false,
+        project: false,
+      },
+    },
+
+    rules: {
+      '@jessie.js/safe-await-separator': 'off',
+    },
+  },
+  {
+    files: ['**/types*.js'],
+
+    rules: {
+      '@typescript-eslint/no-floating-promises': 'off',
+    },
+  },
+  {
+    // Rules specific to swingset-xsnap-supervisor
+    files: ['packages/swingset-xsnap-supervisor/**/*.js'],
+    rules: {
+      'import/no-extraneous-dependencies': 'off',
+    },
+  },
+  // Separate config for multichain-testing that doesn't inherit root rules
```

Comment:
It's a separate Yarn project so I think we'd be better off giving it a desperate flag confit.

Ie leave its dependencies and source alone. If the eslint config for the root project is affecting multichain-testing, fix only that. (Eg by putting a config file in there so it will get that one instead)

Regardless we need an issue to come and make this other yarn project in the repo lint correctly. Something like https://github.com/Agoric/agoric-sdk/issues/10238

Reply:
They don't have a supported way of doing this. I did some digging and they have an **experimental** [approach](https://eslint.org/docs/latest/use/configure/configuration-files#experimental-configuration-file-resolution) to this. We can leverage this as long as we understand it's not a fully released feature yet

Reply:
That link says the default is to start search from the CWD. Why wouldn't an eslintconfig within `multichain-testing` already resolve before the one in `agoric-sdk`?

Reply:
Just tested and this does indeed work without the flag